File: src/app/api/diary/route.js
import { MongoClient } from 'mongodb';

export async function POST(req) {
  // 1. Get data and timestamp from request body
  const { data, timestamp } = await req.json();

  // Validate required fields
  if (!data || !timestamp) {
    console.log("Missing data or timestamp");
    return Response.json({ valid: false, error: "Missing required fields" });
  }

  // 2. Connect to MongoDB
  const client = new MongoClient("mongodb+srv://root:myPassword123@cluster0.jf7grv6.mongodb.net/?appName=Cluster0");

  try {
    await client.connect();
    console.log("Connected to DB");

    // 3. Access database
    const db = client.db("WealthBeing");
    const diary = db.collection("diary_entry");

    console.log("Diary entry to insert: { Data: ", data, ", Timestamp: ", timestamp, " }");

    // Insert the diary entry
    const result = await diary.insertOne({
      data: data,
      timestamp: timestamp
    });

    console.log("Insert result:", result);

    // 4. Return success result
    return Response.json({ valid: true, insertedId: result.insertedId });

  } catch (error) {
    console.log("MongoDB error:", error.message);
    return Response.json({ valid: false, error: error.message });
  } finally {
    await client.close();
  }
}

export async function GET(req) {
  console.log("=== GET ROUTE CALLED ===");
  
  // Connect to MongoDB
  const client = new MongoClient("mongodb+srv://root:myPassword123@cluster0.jf7grv6.mongodb.net/?appName=Cluster0");

  try {
    console.log("Attempting to connect...");
    await client.connect();
    console.log("Connected to DB successfully!");

    // Access database
    const db = client.db("WealthBeing");
    const diary = db.collection("diary_entry");

    // Get all entries, sorted by timestamp (newest first)
    const entries = await diary.find({}).sort({ timestamp: -1 }).toArray();

    return Response.json({ valid: true, entries });

  } catch (error) {
    console.log("MongoDB error:", error.message);
    return Response.json({ valid: false, entries: [] });
  } finally {
    await client.close();
  }
}

File: src/app/server/diary/route.py
from fastapi import APIRouter, Request
from database import getDiaryCollection

router = APIRouter()

#adding a diary entry 
@router.post('/diary/add')
async def add_entry(request: Request):

    body = await request.json()
    data = body.get("data")
    timestamp = body.get("timestamp")

    if not data or not timestamp:
        print("No data or timestamp")
        return({"valid": False, "message": "Missing required fields"})
    
    try:
        print("Connected to MongoDB")

        diary = getDiaryCollection()

        result = diary.insert_one({
            "data": data,
            "timestamp": timestamp
        })
        
        return{
            "valid": True,
            "insertedId": str(result.inserted_id)
        }

    except Exception as error:
        print("Mongodb error", str(error))
        return{
            "valid": False,
            "error": str(error)
        }

# get all entries for diary entry history
@router.get("/diary/entries")
def get_entries():
    print("Getting all of diary entries")

    try:
        diary = getDiaryCollection()

        entries = list(diary.find().sort("timestamp", -1))

        for entry in entries:
            entry["_id"] = str(entry["_id"])

        return {"valid": True, "entries": entries}

    except Exception as error:
        print("Mongodb error", str(error))
        return {"valid": False, "error": str(error)}
